osu.Game/Database/EFToRealmMigrator.cs

```diff
@@ -30,97 +33,119 @@ public EFToRealmMigrator(DatabaseContextFactory efContextFactory, RealmContextFa
-
-                migrateBeatmaps(db);
-                migrateScores(db);
+                migrateSettings(ef);
+                migrateSkins(ef);
+                migrateBeatmaps(ef);
+                migrateScores(ef);
             }
+
+            // Delete the database permanently.
+            // Will cause future startups to not attempt migration.
+            Logger.Log("Migration successful, deleting EF database", LoggingTarget.Database);
+            efContextFactory.ResetDatabase();
         }
 
-        private void migrateBeatmaps(DatabaseWriteUsage db)
+        private void migrateBeatmaps(DatabaseWriteUsage ef)
         {
             // can be removed 20220730.
-            var existingBeatmapSets = db.Context.EFBeatmapSetInfo
-                                        .Include(s => s.Beatmaps).ThenInclude(b => b.RulesetInfo)
-                                        .Include(s => s.Beatmaps).ThenInclude(b => b.Metadata)
-                                        .Include(s => s.Beatmaps).ThenInclude(b => b.BaseDifficulty)
-                                        .Include(s => s.Files).ThenInclude(f => f.FileInfo)
-                                        .Include(s => s.Metadata)
-                                        .ToList();
+            List<EFBeatmapSetInfo> existingBeatmapSets = ef.Context.EFBeatmapSetInfo
+                                                           .Include(s => s.Beatmaps).ThenInclude(b => b.RulesetInfo)
+                                                           .Include(s => s.Beatmaps).ThenInclude(b => b.Metadata)
+                                                           .Include(s => s.Beatmaps).ThenInclude(b => b.BaseDifficulty)
+                                                           .Include(s => s.Files).ThenInclude(f => f.FileInfo)
+                                                           .Include(s => s.Metadata)
+                                                           .ToList();
+
+            Logger.Log("Beginning beatmaps migration to realm", LoggingTarget.Database);
 
             // previous entries in EF are removed post migration.
             if (!existingBeatmapSets.Any())
+            {
+                Logger.Log("No beatmaps found to migrate", LoggingTarget.Database);
                 return;
+            }
 
             using (var realm = realmContextFactory.CreateContext())
-            using (var transaction = realm.BeginWrite())
             {
+                Logger.Log($"Found {existingBeatmapSets.Count} beatmaps in EF", LoggingTarget.Database);
+                string migration = $"before_beatmap_migration_{DateTimeOffset.UtcNow.ToUnixTimeSeconds()}";
+                efContextFactory.CreateBackup($"client.{migration}.db");
+
                 // only migrate data if the realm database is empty.
                 // note that this cannot be written as: `realm.All<BeatmapInfo>().All(s => s.Protected)`, because realm does not support `.All()`.
```

Comment:
nit: outdated comment

```suggestion
```

Reply:
Due to the `BeatmapInfo` instead of `BeatmapSetInfo`?